verified-reviews/loaders/productList.ts

```diff
@@ -0,0 +1,53 @@
+import { AppContext } from "../mod.ts";
+import { AggregateRating, Product } from "../../commerce/types.ts";
+import { ExtensionOf } from "../../website/loaders/extension.ts";
+import { Ratings } from "../utils/types.ts";
+import { createClient, getProductId } from "../utils/client.ts";
+
+/**
+ * @title Opiniões verificadas - Ratings for Products[]
+ */
+export default function productList(
+	config: unknown,
+	req: Request,
+	ctx: AppContext
+): ExtensionOf<Product[] | null> {
+	const client = createClient({ ...ctx });
+
+	return async (products: Product[] | null) => {
+		if (!products) {
+			return null;
+		}
+		if (!client) return products;
+
+		const productsIds = products.map(getProductId);
+		const ratings = await client.ratings({ productsIds: productsIds });
+
+		return products.map((product) => {
+			const productId = getProductId(product);
+			product.aggregateRating = getRatingProduct({
+				ratings,
+				productId: productId,
+			});
+			return product;
+		});
+	};
+}
+
+const getRatingProduct = ({
+	ratings,
+	productId,
+}: {
+	ratings: Ratings | undefined;
+	productId: string;
+}): AggregateRating | undefined => {
+	if (!ratings) return undefined;
+	const rating = ratings[productId]?.[0];
+	if (!rating) return undefined;
```

Comment:
you can simplify by checking it once
```suggestion
	const rating = ratings?.[productId]?.[0];
	if (!rating) { 
	    return undefined; 
	 }
```